src/server/server.ts

```diff
@@ -598,7 +598,9 @@ namespace ts.server {
             if (option && value) {
                 switch (option) {
                     case "-file":
-                        logEnv.file = stripQuotes(value);
+                        const logPathAndNumberOfExtraParts = getLogPath(args, i + 1);
+                        logEnv.file = logPathAndNumberOfExtraParts[0];
+                        i += logPathAndNumberOfExtraParts[1];

```

Comment:
You may want to familiarize yourself with destructuring in ES6+ (http://exploringjs.com/es6/ch_destructuring.html#sec_multiple-return-values)

Reply:
This is exactly what I wanted to do. Thanks for the pointer.